core/pkg/sync/builder/syncbuilder.go

```diff
@@ -27,6 +29,7 @@ const (
 	syncProviderGrpc       = "grpc"
 	syncProviderKubernetes = "kubernetes"
 	syncProviderHTTP       = "http"
+	syncProviderGcs        = "gcs"
```

Comment:
The implementation seems to work for other object storages as well (would require registering their drivers). Shall we rename it to something like `blob` or `objectStorage`?

Reply:
GCS sync provider uses blob as implementation but has it's own uri format.
The flagd users don't need to know what is the internal implementation.